scripts/flow_track_correl.py

```diff
@@ -0,0 +1,414 @@
+
+    return avg_speeds, sigmas
+
+
+def plot_flow_map(speeds: np.ndarray, sigmas: np.ndarray, ycen_band_rs: np.ndarray, rbands: list[int], nlatbins: int,
+                  cmap: str = 'inferno'):
+    """
+    Plot polar maps of the radial flows.
+
+    Parameters
+    ----------
+    speeds : np.ndarray
+        Averaged speed over each radial band and latitudinal bin.
+    sigmas : np.ndarray
+        1-sigma uncertainty associated with each binned speed.
+    ycen_band_rs : np.ndarray
+        y-coordinates of center of bands in solar radii.
+    rbands : list[int]
+        Indices of the radial bands to visualize.
+    nlatbins : int
+        Number of angular bins over 360 degrees.
+    cmap : str, optional
+        Colormap for the plot (default is 'inferno').
+    """
+    thetas = np.linspace(0, 2 * np.pi, nlatbins + 1)
+
+    plt.close('all')
+    fig = plt.figure(figsize=(20, 8))
+
+    vmin = speeds.min()
+    vmax = speeds.max()
+    norm = matplotlib.colors.Normalize(vmin=vmin, vmax=vmax, clip=True)
+    mapper = cm.ScalarMappable(norm=norm, cmap=cmap)
+
+    for i, ridx in enumerate(rbands):
+
+        signal = np.append(speeds[ridx], speeds[ridx][0])
+        error = np.append(sigmas[ridx], sigmas[ridx][0])
+
+        ax = fig.add_subplot(1, len(rbands), i + 1, projection='polar')
+        ax.plot(thetas, signal, 'k-')
+        ax.fill_between(thetas, signal - error, signal + error, alpha=0.3, color='gray')
+
+        colors = np.array([mapper.to_rgba(v) for v in signal])
+        for theta, value, err, color in zip(thetas, signal, error, colors):
+            ax.plot(theta, value, 'o', color=color, ms=4)
+            ax.errorbar(theta, value, yerr=err, lw=2, capsize=3, color=color)
+
+        ax.set_title(f'Altitude = {ycen_band_rs[ridx]} Rs')
+        ax.set_ylim(50, 1.05 * vmax)
+        ax.set_rlabel_position(270)
+
+    cbar_ax = fig.add_axes([0.11, 0.2, 0.8, 0.03])
+    plt.colorbar(mapper, cax=cbar_ax, orientation='horizontal').ax.set_xlabel('Speed (km/s)')
+    plt.savefig('Radial_Speed_Map.png')
+
+
+if __name__ == "__main__":
+    # Input parameters and configuration
+    files = sorted(glob.glob(os.path.join(os.environ['PUNCHDATA'], '*')))
+    hdu = 0  # Index of the header data unit (change if necessary)
+    dt = 12  # Time offset in frames between images
+    sparsity = 2  # Frame skip interval for averaging
+    n_ofs = 151  # Number of spatial offsets for cross-correlation
+    dr = 2  # Pixel offset increment per sample
+    of0 = dt * 9  # Central offset based on expected wind speed (300 km/s)
+    expected_kps_windspeed = 300  # Expected wind speed in km/s
+    chw = 0.5  # Half-width of each radial band in solar radii
+    nlatbins = 36  # Number of angular bins (latitude bins)
+    # Define radial band centers in solar radii
+    ycens = np.array([7, 7.5, 8, 8.5, 9, 9.5, 10, 10.5, 11, 11.5, 12, 12.5, 13, 13.5, 14])
+    rbands = [0, 4, 8, 14]  # Indices of radial bands to visualize
```

Comment:
I'd like to understand these parameters a bit better. I also lean towards the side of longer, more explicit variable names so it's clear what they mean when they're used. For example, `chw` might be changed to `radial_half_width`. Those are pretty straight forward refactorings once we have everything working right.

Reply:
Also, what if we have an image missing because we couldn't downlink from the satellite or something? This seems to assume the images are regularly spaced. Would it be helpful to use the observation time from the header?

Reply:
These parameter list are COR2-related and the intent is that once I have the mosaic data with the full extent of the field of view, I would like to reduce the numbers of parameters that we need to input. In any case, I can change their names for longer names, I also prefer more explicit, longer names for the sake of quickly understanding what's going on without having to look at the documentation.